lib/cardano-api-extra/lib/Cardano/Api/Gen.hs

```diff
@@ -1172,13 +1175,15 @@ genUnsignedQuantity = do
         ]
 
 genTxOutValue :: CardanoEra era -> Gen (TxOutValue era)
-genTxOutValue era = case forEraMaybeEon era of
-    Nothing -> TxOutValueByron <$> genLovelace
-    Just (_supported :: ShelleyBasedEra era) ->
-        oneof
-            [ TxOutValueByron <$> genLovelace
-            , genTxOutValue era
-            ]
+genTxOutValue era = case era of
+    ConwayEra -> TxOutValueShelleyBased ShelleyApi.ShelleyBasedEraConway
+            . Api.toLedgerValue @ConwayEra MaryEraOnwardsConway
+            <$> genValueForTxOut
+    BabbageEra -> TxOutValueShelleyBased ShelleyApi.ShelleyBasedEraBabbage
+            . Api.toLedgerValue @BabbageEra MaryEraOnwardsBabbage
+            <$> genValueForTxOut
+
+    _ -> error "genTxOutValue: unsupported era"
```

Comment:
I don't think there is a strong need in this case, as:
- I really hope we rewrite these generators long before the next era
- If not: worst thing that happens we'll notice the problems when running the tests instead of when compiling